### ISSUE TITLE
App manager doesn't run if you relaunch it.

### ISSUE THREAD

**Issue by vordx** [MuntashirAkon/AppManager#1355]
### Please check before submitting an issue

- [X] I know what my device, OS and App Manager versions are
- [X] I know how to take logs
- [X] I know how to reproduce the issue which may not be specific to my device

### Describe the bug

App manager gets stuck when relaunching it from the background

### To Reproduce

https://github.com/MuntashirAkon/AppManager/assets/39927714/4cbaf905-c6eb-4e5e-bfc5-3b5e4f6f86b8

### Expected behavior

Launch fine?

### Screenshots

Shared a video above

### Device info

- Device: Galaxy s23U
- OS Version: Android 14, oneui 6.1
- App Manager Version: 4.0.0-beta01 (436)
- Mode: ADB

**Comment by ascendbeing** [comment]
having this issue too. samsung a13 5g. February Samsung ota security update is possibly related because after installing it i could no longer view appdata in mixplorer. only after mixplorer added shizuku to beta features could i manage other apps appdatas, after updating to that February security update.
the symptom for me is as you say. i cannot rerun am in adb mode until i reboot. if i launch it, it won't say wireless debugging connected. it'll just sit there until either it times out (takes ages, like 5-10 min of waiting? doing nothing else, cause if you switch away it doesn't time out from background) or if it does time out it'll prompt which mode you want.
neither wirelessly debugging mode nor adb over tcp fix this. they both have this happen, and the only resolution is reboot the device.
i have even tried doing stuff like disabling phantom process monitoring and making sure shizuku isn't running. nothing fixes it. i think clearing app data of AM MIGHT. but that's a pretty extreme measure to have to take just to get am to launch in a manner that allows to change app ops and perms.

**Comment by ascendbeing** [comment]
![Screenshot_20240528_032418_Firefox Nightly](https://github.com/MuntashirAkon/AppManager/assets/119428197/f09e7d49-4906-45ea-a261-e2f3f3c665ec)
i guess that sometimes it will pop up and tell you it failed to grab adb mode. this is android 13 the device is sm-a136.

**Comment by ascendbeing** [comment]
i didnt have am stable installed so i installed the latest and i was able to launch am 3.1.6 adb mode after pairing. still took ages to let me switch modes or choose pair etc. but it works. OP. maybe try that. i mostly use the snapshot version. the snapshot is the one that's having the issue, i dont remember it happening on 2 or 3.x but I'll share if it does.

**Comment by MuntashirAkon** [comment]
@vordx: Can you take some logs when you try to launch it from background using the debug build? According to your video, it launched fine, but it didn't list any apps.

@ascendbeing: Please create a new issue with everything you have on the issue. Make sure to explain it in a step by step manner so that I can follow you. Please refrain from indicating what may went wrong other than the ones you're sure about (e.g., February updates, etc.). This creates a bias which makes it difficult to find out the actual issue.

**Comment by ascendbeing** [comment]
@MuntashirAkon although I am not the person who you asked for logs, here, I have some for you. This is what I see in logcat after I've previously ran AM and it worked, and this log is triggered either when trying to relaunch it (when it tries to go adb but fails) or when I got it on non root mode since that didn't work and am trying to change it to "auto" while wireless debugging is currently enabled in either case;
```06-08 06:51:36.037 31647 31647 W System.err: android.view.WindowManager.LayoutParams Unable to load class java.lang.ClassNotFoundException: android.view.WindowManager.LayoutParams                                                         06-08 06:52:04.448 31588 31639 W System.err: java.net.SocketTimeoutException: Read timed out                          06-08 06:52:04.450 31588 31639 W System.err:    at java.net.SocketInputStream.socketRead0(Native Method)              06-08 06:52:04.450 31588 31639 W System.err:    at java.net.SocketInputStream.socketRead(SocketInputStream.java:118)  06-08 06:52:04.450 31588 31639 W System.err:    at java.net.SocketInputStream.read(SocketInputStream.java:173)        06-08 06:52:04.450 31588 31639 W System.err:    at java.net.SocketInputStream.read(SocketInputStream.java:143)        06-08 06:52:04.450 31588 31639 W System.err:    at java.io.DataInputStream.readFully(DataInputStream.java:203)        06-08 06:52:04.450 31588 31639 W System.err:    at java.io.DataInputStream.readInt(DataInputStream.java:394)          06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.server.common.DataTransmission.readMessage(DataTransmission.java:131)                      06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.server.common.DataTransmission.sendAndReceiveMessage(DataTransmission.java:149)            06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.servermanager.LocalServerManager.execPre(LocalServerManager.java:148)                      06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.servermanager.LocalServerManager.execNew(LocalServerManager.java:160)                      06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.servermanager.LocalServer.exec(LocalServer.java:140)                                       06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.servermanager.LocalServer.runCommand(LocalServer.java:128)                                 06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.ipc.RootService.lambda$asRunnable$0(RootService.java:203)                                  06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.ipc.RootService$$ExternalSyntheticLambda0.run(D8$$SyntheticClass:0)                        06-08 06:52:04.450 31588 31639 W System.err:    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)                                               06-08 06:52:04.450 31588 31639 W System.err:    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)                                               06-08 06:52:04.450 31588 31639 W System.err:    at java.lang.Thread.run(Thread.java:1012)                             06-08 06:53:23.578 31588 31961 W System.err: java.io.FileNotFoundException: Cannot find file:///data/system/users/0/settings_ssaid.xml                                           06-08 06:53:23.578 31588 31961 W System.err:    at io.github.muntashirakon.io.PathImpl.findFile(PathImpl.java:469)    06-08 06:53:23.578 31588 31961 W System.err:    at io.github.muntashirakon.AppManager.ssaid.SsaidSettings.<init>(SsaidSettings.java:51)                                          06-08 06:53:23.578 31588 31961 W System.err:    at io.github.muntashirakon.AppManager.db.utils.AppDb.updateVariableData(AppDb.java:374)                                          06-08 06:53:23.578 31588 31961 W System.err:    at io.github.muntashirakon.AppManager.db.utils.AppDb.updateApplications(AppDb.java:295)                                          06-08 06:53:23.578 31588 31961 W System.err:    at io.github.muntashirakon.AppManager.db.utils.AppDb.loadInstalledOrBackedUpApplications(AppDb.java:151)                         06-08 06:53:23.578 31588 31961 W System.err:    at io.github.muntashirakon.AppManager.utils.PackageUtils.lambda$getInstalledOrBackedUpApplicationsFromDb$0(PackageUtils.java:253)06-08 06:53:23.578 31588 31961 W System.err:    at io.github.muntashirakon.AppManager.utils.PackageUtils$$ExternalSyntheticLambda2.run(D8$$SyntheticClass:0)                     06-08 06:53:23.578 31588 31961 W System.err:    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:487)                                                       06-08 06:53:23.578 31588 31961 W System.err:    at java.util.concurrent.FutureTask.run(FutureTask.java:264)           06-08 06:53:23.578 31588 31961 W System.err:    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)                                               06-08 06:53:23.578 31588 31961 W System.err:    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)                                               06-08 06:53:23.578 31588 31961 W System.err:    at java.lang.Thread.run(Thread.java:1012)                             06-08 06:53:37.437 31588 31965 W System.err: java.io.FileNotFoundException: Cannot find file:///data/system/users/0/settings_ssaid.xml                                           06-08 06:53:37.437 31588 31965 W System.err:    at io.github.muntashirakon.io.PathImpl.findFile(PathImpl.java:469)    06-08 06:53:37.437 31588 31965 W System.err:    at io.github.muntashirakon.AppManager.ssaid.SsaidSettings.<init>(SsaidSettings.java:51)                                          06-08 06:53:37.437 31588 31965 W System.err:    at io.github.muntashirakon.AppManager.db.utils.AppDb.updateVariableData(AppDb.java:374)                                          06-08 06:53:37.437 31588 31965 W System.err:    at io.github.muntashirakon.AppManager.db.utils.AppDb.updateApplications(AppDb.java:295)                                          06-08 06:53:37.438 31588 31965 W System.err:    at io.github.muntashirakon.AppManager.db.utils.AppDb.loadInstalledOrBackedUpApplications(AppDb.java:151)                         06-08 06:53:37.438 31588 31965 W System.err:    at io.github.muntashirakon.AppManager.utils.PackageUtils.lambda$getInstalledOrBackedUpApplicationsFromDb$0(PackageUtils.java:253)06-08 06:53:37.438 31588 31965 W System.err:    at io.github.muntashirakon.AppManager.utils.PackageUtils$$ExternalSyntheticLambda2.run(D8$$SyntheticClass:0)                     06-08 06:53:37.438 31588 31965 W System.err:    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:487)                                                       06-08 06:53:37.438 31588 31965 W System.err:    at java.util.concurrent.FutureTask.run(FutureTask.java:264)           06-08 06:53:37.438 31588 31965 W System.err:    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)                                               06-08 06:53:37.438 31588 31965 W System.err:    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)                                               06-08 06:53:37.438 31588 31965 W System.err:    at java.lang.Thread.run(Thread.java:1012)```

**Comment by MuntashirAkon** [comment]
> 06-08 06:52:04.448 31588 31639 W System.err: java.net.SocketTimeoutException: Read timed out                          
> 06-08 06:52:04.450 31588 31639 W System.err:    at java.net.SocketInputStream.socketRead0(Native Method)              
> 06-08 06:52:04.450 31588 31639 W System.err:    at java.net.SocketInputStream.socketRead(SocketInputStream.java:118)  
> 06-08 06:52:04.450 31588 31639 W System.err:    at java.net.SocketInputStream.read(SocketInputStream.java:173)        
> 06-08 06:52:04.450 31588 31639 W System.err:    at java.net.SocketInputStream.read(SocketInputStream.java:143)        
> 06-08 06:52:04.450 31588 31639 W System.err:    at java.io.DataInputStream.readFully(DataInputStream.java:203)        
> 06-08 06:52:04.450 31588 31639 W System.err:    at java.io.DataInputStream.readInt(DataInputStream.java:394)          
> 06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.server.common.DataTransmission.readMessage(DataTransmission.java:131)                      
> 06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.server.common.DataTransmission.sendAndReceiveMessage(DataTransmission.java:149)            
> 06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.servermanager.LocalServerManager.execPre(LocalServerManager.java:148)                      
> 06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.servermanager.LocalServerManager.execNew(LocalServerManager.java:160)                      
> 06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.servermanager.LocalServer.exec(LocalServer.java:140)                                       
> 06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.servermanager.LocalServer.runCommand(LocalServer.java:128)                                 
> 06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.ipc.RootService.lambda$asRunnable$0(RootService.java:203)                                  
> 06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.ipc.RootService$$ExternalSyntheticLambda0.run(D8$$SyntheticClass:0)                        
> 06-08 06:52:04.450 31588 31639 W System.err:    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)                                               
> 06-08 06:52:04.450 31588 31639 W System.err:    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)                                               
> 06-08 06:52:04.450 31588 31639 W System.err:    at java.lang.Thread.run(Thread.java:1012)

This part looks interesting.

You need to be more thorough about your report. Instead of saying everything in one complex line, try to write it in an ordered list. This way I can understand the situation better and figure out what may go wrong. ADB is all about interacting with the `adbd` demon, and depending on operating system, a lot of things can go wrong. I would also suggest installing a debug build and send all the logs when it fails. You can also send me the logs privately at am4android-at-riseup-dot-net.

**Comment by ascendbeing** [comment]
> > 06-08 06:52:04.448 31588 31639 W System.err: java.net.SocketTimeoutException: Read timed out
> > 06-08 06:52:04.450 31588 31639 W System.err:    at java.net.SocketInputStream.socketRead0(Native Method)
> > 06-08 06:52:04.450 31588 31639 W System.err:    at java.net.SocketInputStream.socketRead(SocketInputStream.java:118)
> > 06-08 06:52:04.450 31588 31639 W System.err:    at java.net.SocketInputStream.read(SocketInputStream.java:173)
> > 06-08 06:52:04.450 31588 31639 W System.err:    at java.net.SocketInputStream.read(SocketInputStream.java:143)
> > 06-08 06:52:04.450 31588 31639 W System.err:    at java.io.DataInputStream.readFully(DataInputStream.java:203)
> > 06-08 06:52:04.450 31588 31639 W System.err:    at java.io.DataInputStream.readInt(DataInputStream.java:394)
> > 06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.server.common.DataTransmission.readMessage(DataTransmission.java:131)
> > 06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.server.common.DataTransmission.sendAndReceiveMessage(DataTransmission.java:149)
> > 06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.servermanager.LocalServerManager.execPre(LocalServerManager.java:148)
> > 06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.servermanager.LocalServerManager.execNew(LocalServerManager.java:160)
> > 06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.servermanager.LocalServer.exec(LocalServer.java:140)
> > 06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.servermanager.LocalServer.runCommand(LocalServer.java:128)
> > 06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.ipc.RootService.lambda$asRunnable$0(RootService.java:203)
> > 06-08 06:52:04.450 31588 31639 W System.err:    at io.github.muntashirakon.AppManager.ipc.RootService$$ExternalSyntheticLambda0.run(D8$$SyntheticClass:0)
> > 06-08 06:52:04.450 31588 31639 W System.err:    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
> > 06-08 06:52:04.450 31588 31639 W System.err:    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
> > 06-08 06:52:04.450 31588 31639 W System.err:    at java.lang.Thread.run(Thread.java:1012)
> 
> This part looks interesting.
> 
> You need to be more thorough about your report. Instead of saying everything in one complex line, try to write it in an ordered list. This way I can understand the situation better and figure out what may go wrong. ADB is all about interacting with the `adbd` demon, and depending on operating system, a lot of things can go wrong. I would also suggest installing a debug build and send all the logs when it fails. You can also send me the logs privately at am4android-at-riseup-dot-net.
@MuntashirAkon 
I am tentatively saying your recent changes wrt Samsung may have fixed every aspect of this. I was able to relaunch AM debug, for example. So the titular issue APPEARS tentatively resolved. So thank you for making major progress on this. It was a very annoying thing. My main concern was that less tech inclined people than I would understand it much less than I, which would have the domino effect, of this project being slept on. So I have a very positive outlook on AM as a result of your recent issue sweep. Thanks for taking the time.

**Comment by MuntashirAkon** [comment]
> I am tentatively saying your recent changes wrt Samsung may have fixed every aspect of this.

Probably a side-effect of 93ce56cc89f9fc4fd109098370b933ff1f1bb430

> My main concern was that less tech inclined people than I would understand it much less than I, which would have the domino effect, of this project being slept on.

Tech-savvy people with interests in privacy are the target audience of this project. This is not just a motto, it's a tech limit as well. App Manager relies heavily on the framework itself which is revised very frequently and differently in each device. So, the up-to-date users should always expect a certain level of instability with the app.

**Comment by MuntashirAkon** [comment]
Closing this issue for now.

**Comment by ascendbeing** [comment]
@MuntashirAkon I think we are talking past each other. Please just make sure the fixes you made to resolve this in HEAD also get cherry picked and put into a forthcoming (or already out?) stable release. Because when I provided you the logs, I couldn't relaunch w/ auto adb or wireless debug or tcp on: a) HEAD b) latest stable c) arbitrarily chosen old stable.

If you already did that, cool. If not, it kinda flies against the idea, that it's the users on HEAD, who should be the ones, seeing instability.
If you need logs, let me know. I can email you whichever, if any, you'd like checked (latest release or latest stable branch push). I don't know if you back ported the fix. I'm just specifically addressing your point that it's people on HEAD who are the ones intended for the less stable experience.

**Comment by MuntashirAkon** [comment]
> @MuntashirAkon I think we are talking past each other. Please just make sure the fixes you made to resolve this in HEAD also get cherry picked and put into a forthcoming (or already out?) stable release. Because when I provided you the logs, I couldn't relaunch w/ auto adb or wireless debug or tcp on: a) HEAD b) latest stable c) arbitrarily chosen old stable.
> 
> If you already did that, cool. If not, it kinda flies against the idea, that it's the users on HEAD, who should be the ones, seeing instability. If you need logs, let me know. I can email you whichever, if any, you'd like checked (latest release or latest stable branch push). I don't know if you back ported the fix. I'm just specifically addressing your point that it's people on HEAD who are the ones intended for the less stable experience.

The OP has reported the issue from the latest beta version. So, it was assumed it only affected the beta version.

**Comment by MuntashirAkon** [comment]
However, this particular change has been back ported.
